### DISCUSSION TITLE
Nested dependecies

### DISCUSSION THREAD

**Discussion by DryganDev** [pointfreeco/swift-dependencies#234] [Q&A]
I want to make Client as Dependency in TCA and I want to add Dependency to Client to make it more flexible. Unfortunately, I can do that according to the guide. Any ideas ho to fix it?

<img width="997" alt="Screenshot 2024-06-28 at 22 15 06" src="https://github.com/pointfreeco/swift-dependencies/assets/5908760/e4fc89ae-a1ff-4aee-a71d-bc9bd3eced7b">

**Comment by Nikolaiko** [comment]
Hi, @DryganDev I was wondering about same question. And looked through discussions. Looks like you can fix it if you move dependency inside your creating closure Self {....}. Example is [here](https://github.com/pointfreeco/swift-dependencies/discussions/182#discussioncomment-8942203). Worked for me.

But maybe developers will correct me if I am wrong.

**Reply by mbrandonw** [reply]
Yes that is correct. You can use `@Dependency(\.remoteClient)` directly in the `getItems` and `pushItems` endpoints.

**Reply by acosmicflamingo** [reply]
Within the context of testing TCA stores and plucking dependencies from the ether in endpoints, is it a big deal if the test suite only recognizes that `itemClient` is being used but not `remoteClient`? I'm guessing the answer is no because it could get overly complicated to have to declare dependencies of dependencies. Am I right?

**Reply by acosmicflamingo** [reply]
For example, in my tests I just found that one of my models marked `l11yClient.language` as unimplemented, but I was surprised to see that `locale` was not marked as unimplemented, even though `l11yClient.language` has to pluck some data from the `locale` dependency.
